### PR TITLE
Add Antrea Operator boiler plate code

### PR DESCRIPTION
**Pull request by ruicao93**
1. Use operator-sdk to create Antrea operator:
$ operator-sdk new antrea-operator
2. Use operator-sdk to add configmap, pod controllers:
$operator-sdk add controller --api-version=k8s.io/v1 --kind=ConfigMap --custom-api-import=k8s.io/api/core/v1
$operator-sdk add controller --api-version=k8s.io/v1 --kind=Pod --custom-api-import=k8s.io/api/core/v1

### CODE REVIEW

**Comment by dantingl** [comment]
> 2. Use operator-sdk to add configmap, pod, node controllers:

remove node

File: deploy/role.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: antrea-operator
+rules:
+- apiGroups:
+  - ""
+  resources:
```
Comment (dantingl) [inline deploy/role.yaml:8]: How about we only add what we need for now, and add more later?

File: deploy/operator.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: antrea-operator
```
Comment (dantingl) [inline deploy/operator.yaml:4]: Do you think we should have a separate namespace for operator, or we just use kube-system?

Comment (dantingl) [reply deploy/operator.yaml:4]: Let's create a separate antrea-operator namespace for now

Comment (ruicao93) [reply deploy/operator.yaml:4]: Done. Set  namespace as `antrea-operator`.

File: deploy/role.yaml (hunk 1 above)
Comment (ruicao93) [reply deploy/role.yaml:8]: Let's add rules when need them.

File: deploy/operator.yaml (hunk 2 above)
Comment (dantingl) [reply deploy/operator.yaml:4]: You also need a namespace yaml file

Comment (ruicao93) [reply deploy/operator.yaml:4]: Done.
- Add namespace yml
- Update role type as `ClusterRole`

**Comment by ruicao93** [comment]
> > 1. Use operator-sdk to add configmap, pod, node controllers:
> 
> remove node

Done.

**Review by dantingl — APPROVED** [review]

**Pull request merged**